zune-png/src/utils.rs

```diff
@@ -350,3 +350,9 @@ pub(crate) fn add_alpha(input: &[u8], output: &mut [u8], colorspace: PngColor, d
         (a, b) => panic!("Unknown combination of depth {a:?} and color type for expand alpha {b:?}")
     }
 }
+
+pub fn convert_u16_to_u8_slice(slice: &mut [u16]) -> &mut [u8] {
+    // Converting a u16 slice to a u8 slice is always correct because
+    // the alignment of the target is smaller.
+    unsafe { std::slice::from_raw_parts_mut(slice.as_ptr() as *mut u8, slice.len() * 2) }
```

Comment:
What if `slice.len() * 2` overflows?

Reply:
I don't think it can but there should be a comment explaining why it cannot

Reply:
I also thought it can not, but fixed it with a `checked_mul`.